src/UglyToad.PdfPig/Geometry/PdfPath.cs

```diff
@@ -638,6 +714,38 @@ internal void Rectangle(decimal x, decimal y, decimal width, decimal height)
             LineTo(x + width, y + height);
             LineTo(x, y + height);
             LineTo(x, y);
+            IsDrawnAsRectangle = true;
+        }
+
+        /// <summary>
+        /// Order matters
+        /// </summary>
+        public override bool Equals(object obj)
+        {
+            if (obj is PdfPath path)
+            {
+                if (this.Commands.Count != path.Commands.Count) return false;
+
+                for (int i = 0; i < this.Commands.Count; i++)
+                {
+                    if (!this.Commands[i].Equals(path.Commands[i])) return false;
+                }
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Order matters
+        /// </summary>
+        public override int GetHashCode()
+        {
+            var hash = this.Commands.Count + 1;
+            for (int i = 0; i < this.Commands.Count; i++)
+            {
+                hash = hash * (i + 1) + this.Commands[i].GetHashCode();
```

Comment:
All the `GetHashCode` implementations in this file look good but this one makes me unsure. I think it's probably 🤞 ok but it might be worth using something from here? https://stackoverflow.com/questions/3404715/c-sharp-hashcode-for-array-of-ints

Reply:
I've modified it slightly but I basically kept the same logic (cf accepted answer in your link). Let me know if you think of something else :)